s3-native-client/src/main/java/com/amazonaws/s3/S3NativeClient.java

```diff
@@ -79,30 +122,32 @@ public void onFinished(int errorCode) {
                 } finally {
                     if (ex != null) {
                         resultFuture.completeExceptionally(ex);
-                    }
-                    else {
+                    } else {
                         resultFuture.complete(getObjectOutput);
                     }
                 }
             }
         };
 
         List<HttpHeader> headers = new LinkedList<>();
+
         // TODO: additional logic needed for *special* partitions
-        headers.add(new HttpHeader("Host", request.bucket() + ".s3."+ signingRegion + ".amazonaws.com"));
+        headers.add(new HttpHeader("Host", request.bucket() + ".s3." + signingRegion + ".amazonaws.com"));
         populateGetObjectRequestHeaders(header -> headers.add(header), request);
-        final StringBuilder keyString = new StringBuilder("/" + request.key());
+        final StringBuilder queryParameters = new StringBuilder("/" + request.key());
```

Comment:
This feels weird/confusing, query parameters are separate from path and this is kind of mixing them state-wise.

Reply:
I've tried to better manage the state "getEncodedPath" function.